### DISCUSSION TITLE
Gatsby - updating field in Strapi during build time

### DISCUSSION THREAD

**Discussion by MotaZor94** [gatsbyjs/gatsby#27975] [Help] [answered]
Hi everyone,

I am trying to figure out if it's possible to mutate the field in Strapi after the build of the page has been completed.

I am generating 50-60 landing pages which after they are done building get their own path.

For example:

[](https://test.com/newly-generated-page)

What I want to do is I want to send that path back to Strapi every time the user creates a new page and assign it to that page in the database so the user can see it in the "Page Path" field.

I have tried implementing it through gatsby-node.js but I can't seem to find a solution for it.

I've tried something like this with Gatsby's onPostBuild API in gatsby-node.js, I have managed to get a list of all the paths but I am not really sure if this would be the right way of doing it:

```
exports.onPostBuild = ({ graphql }) => {
const getPagePaths = makeRequest(
  graphql,
    `
    {
      allSitePage {
        nodes {
          path 
          id
        }
      }
    }
    `
  ).then(result => {
    result.data.allSitePage.nodes.forEach(({ path, id }) => {
      console.log(path, id)
    })
  })

  return Promise.all([
    getPagePaths
  ])
}
```
Should I maybe use APOLLO for this and do it inside of that specific page template?

Thanks everyone, and have a lovely day!

**Comment by LekoArts** [comment, marked as answer]
Same answer as in https://github.com/gatsbyjs/gatsby/discussions/27997#discussioncomment-130584
